coresimd/aarch64/crypto.rs

```diff
@@ -0,0 +1,428 @@
+pub unsafe fn vsha1cq_u32(
+    hash_abcd: uint32x4_t, hash_e: u32, wk: uint32x4_t
+) -> uint32x4_t {
+    vsha1cq_u32_(hash_abcd, hash_e, wk)
+}
+
+/// SHA1 hash update accelerator, majority.
+#[inline]
+#[target_feature(enable = "crypto")]
+#[cfg_attr(test, assert_instr(sha1m))]
+pub unsafe fn vsha1mq_u32(
+    hash_abcd: uint32x4_t, hash_e: u32, wk: uint32x4_t
+) -> uint32x4_t {
+    vsha1mq_u32_(hash_abcd, hash_e, wk)
+}
+
+/// SHA1 hash update accelerator, parity.
+#[inline]
+#[target_feature(enable = "crypto")]
+#[cfg_attr(test, assert_instr(sha1p))]
+pub unsafe fn vsha1pq_u32(
+    hash_abcd: uint32x4_t, hash_e: u32, wk: uint32x4_t
+) -> uint32x4_t {
+    vsha1pq_u32_(hash_abcd, hash_e, wk)
+}
+
+/// SHA1 schedule update accelerator, first part.
+#[inline]
+#[target_feature(enable = "crypto")]
+#[cfg_attr(test, assert_instr(sha1su0))]
+pub unsafe fn vsha1su0q_u32(
+    w0_3: uint32x4_t, w4_7: uint32x4_t, w8_11: uint32x4_t
+) -> uint32x4_t {
+    vsha1su0q_u32_(w0_3, w4_7, w8_11)
+}
+
+/// SHA1 schedule update accelerator, second part.
+#[inline]
+#[target_feature(enable = "crypto")]
+#[cfg_attr(test, assert_instr(sha1su1))]
+pub unsafe fn vsha1su1q_u32(
+    tw0_3: uint32x4_t, w12_15: uint32x4_t
+) -> uint32x4_t {
+    vsha1su1q_u32_(tw0_3, w12_15)
+}
+
+/// SHA256 hash update accelerator.
+#[inline]
+#[target_feature(enable = "crypto")]
+#[cfg_attr(test, assert_instr(sha256h))]
+pub unsafe fn vsha256hq_u32(
+    hash_abcd: uint32x4_t, hash_efgh: uint32x4_t, wk: uint32x4_t
+) -> uint32x4_t {
+    vsha256hq_u32_(hash_abcd, hash_efgh, wk)
+}
+
+/// SHA256 hash update accelerator, upper part.
+#[inline]
+#[target_feature(enable = "crypto")]
+#[cfg_attr(test, assert_instr(sha256h2))]
+pub unsafe fn vsha256h2q_u32(
+    hash_efgh: uint32x4_t, hash_abcd: uint32x4_t, wk: uint32x4_t
+) -> uint32x4_t {
+    vsha256h2q_u32_(hash_efgh, hash_abcd, wk)
+}
+
+/// SHA256 schedule update accelerator, first part.
+#[inline]
+#[target_feature(enable = "crypto")]
+#[cfg_attr(test, assert_instr(sha256su0))]
+pub unsafe fn vsha256su0q_u32(
+    w0_3: uint32x4_t, w4_7: uint32x4_t
+) -> uint32x4_t {
+    vsha256su0q_u32_(w0_3, w4_7)
+}
+
+/// SHA256 schedule update accelerator, second part.
+#[inline]
+#[target_feature(enable = "crypto")]
+#[cfg_attr(test, assert_instr(sha256su1))]
+pub unsafe fn vsha256su1q_u32(
+    tw0_3: uint32x4_t, w8_11: uint32x4_t, w12_15: uint32x4_t
+) -> uint32x4_t {
+    vsha256su1q_u32_(tw0_3, w8_11, w12_15)
+}
+
+#[cfg(test)]
+mod tests {
+    use stdsimd_test::simd_test;
+    use simd::*;
+    use coresimd::aarch64::*;
+    use std::mem;
+
+    #[simd_test = "crypto"]
+    unsafe fn test_vaeseq_u8() {
+        let data = u8x16::new(1, 2, 3, 4, 5, 6, 7, 8, 1, 2, 3, 4, 5, 6, 7, 8)
+            .into_bits();
+        let key = u8x16::new(0, 1, 2, 3, 4, 5, 6, 7, 0, 1, 2, 3, 4, 5, 6, 7)
+            .into_bits();
+        let r: u8x16 = vaeseq_u8(data, key).into_bits();
+        assert_eq!(
+            r,
+            u8x16::new(
```

Comment:
nitpick: no need to do this, but if you want to manually format these next time just put them in a variable and tell rustfmt to skip these:

```rust
#[cfg_attr(rustfmt, skip)]
let r = u8x16::new(
    1, 2, 3, 4, 5, 6, 7, 8
    9, 10, 11, 12, 13, 14, 15
);
```